### ISSUE TITLE
Zed: Make the reserved words as limited in scope as possible

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2398]
Repro is with `zar` commit e6b1fda.

While working on bringing the `zar` README current (#2266), I bumped into this example in the "simple graph queries" section that's now a parse error:

```
$ zar map -o forward.zng "id.orig_h != null | put from=id.orig_h,to=id.resp_h | count() by from,to"
error parsing Z at column 29:
id.orig_h != null | put from=id.orig_h,to=id.resp_h | count() by from,to
                        === ^ ===
```

I confirmed with @mccanne that this is due to "from" becoming a reserved word in the Z language as part of the recently-added SQL support. However, he also confirmed that we can do better:

> Yes, this should be quite fixable.  In the meantime, you can also use `\from`
> 
> We should make an issue to make the reserved words as limited in scope as possible.

**Comment by philrz** [comment]
I bumped into this again while assembling chanegs for #2448. The problem was in the context of `type` specifications for the reference shaping config for Zeek data. Here were the specific lines from the config that had stopped working when the changes merged for #2215. You can see the backslashes where I had to add escaping.

```
type intel={_path:string,ts:time,uid:bstring,id:id,seen:{indicator:bstring,indicator_type:zenum,\where:zenum,node:bstring},matched:|[zenum]|,sources:|[bstring]|,fuid:bstring,file_mime_type:bstring,file_desc:bstring,_write_ts:time};
type kerberos={_path:string,ts:time,uid:bstring,id:id,request_type:bstring,client:bstring,service:bstring,success:bool,error_msg:bstring,\from:time,till:time,cipher:bstring,forwardable:bool,renewable:bool,client_cert_subject:bstring,client_cert_fuid:bstring,server_cert_subject:bstring,server_cert_fuid:bstring,_write_ts:time};
type smtp={_path:string,ts:time,uid:bstring,id:id,trans_depth:uint64,helo:bstring,mailfrom:bstring,rcptto:|[bstring]|,date:bstring,\from:bstring,to:|[bstring]|,cc:|[bstring]|,reply_to:bstring,msg_id:bstring,in_reply_to:bstring,subject:bstring,x_originating_ip:ip,first_received:bstring,second_received:bstring,last_reply:bstring,path:[ip],user_agent:bstring,tls:bool,fuids:[bstring],is_webmail:bool,_write_ts:time};
```

**Comment by philrz** [comment]
Note to self: For now I've added a comment in the "Custom Brimcap Configuration" article linking to this as an open issue, so that way the reader understands the current limitation is temporary. If/when we address this issue, I should update the wiki article to remove the workaround and the comment.

**Comment by philrz** [comment]
When writing some docs with Zed commit 1f84312, I bumped into a particularly egregious example of this.

These work as expected:

```
$ zq -version
Version: v0.29.0-407-g1f84312c

$ echo '{"Status":"Pending",School: "Wei Yu International Charter"}' | zq -z 'Status=="Pending"' -
{Status:"Pending",School:"Wei Yu International Charter"}

$ echo '{"Status":"Pending",School: "Wei Yu International Charter"}' | zq -z 'international' -
{Status:"Pending",School:"Wei Yu International Charter"}
```

However, when I try to combine the two search terms with the implicit `and`, it no longer matches.

```
$ echo '{"Status":"Pending",School: "Wei Yu International Charter"}' | zq -z 'Status=="Pending" international' -
[no output]
```

A closer look shows the start of `international` is being treated as the reserved word `in`, despite the fact there's no space separator character.

```
$ zc 'Status=="Pending" international'
{
    "kind": "Sequential",
    "procs": [
        {
            "expr": {
                "kind": "BinaryExpr",
                "lhs": {
                    "kind": "BinaryExpr",
                    "lhs": {
                        "kind": "ID",
                        "name": "Status"
                    },
                    "op": "=",
                    "rhs": {
                        "kind": "Primitive",
                        "text": "Pending",
                        "type": "string"
                    }
                },
                "op": "in",
                "rhs": {
                    "kind": "ID",
                    "name": "ternational"
                }
            },
            "kind": "Filter"
        }
    ]
}
```

FWIW, it works as expected with explicit `and`.

```
$ echo '{"Status":"Pending",School: "Wei Yu International Charter"}' | zq -z 'Status=="Pending" and international' -
{Status:"Pending",School:"Wei Yu International Charter"}
```

**Comment by mccanne** [comment]
Just noting that I hit this problem with a field called `order` with this (meta)query...
```
from [pools] | layout.order == "p1"'
```
giving syntax order after the `order` identifier

**Comment by philrz** [comment]
As noted by @mattnibs above, #2968 addressed this one similar-feeling corner case I cited in this issue. Here's the verification of that fix with Zed commit 0f21060. Now the record is successfully matched, since we're no longer seeing the leading `in` as an operator.

```
$ zq -version
Version: v0.29.0-480-g0f21060e

$ echo '{"Status":"Pending",School: "Wei Yu International Charter"}' | zq -z 'Status=="Pending" international' -
{Status:"Pending",School:"Wei Yu International Charter"}

$ zc 'Status=="Pending" international'
{
    "kind": "Sequential",
    "procs": [
        {
            "expr": {
                "kind": "BinaryExpr",
                "lhs": {
                    "kind": "BinaryExpr",
                    "lhs": {
                        "kind": "ID",
                        "name": "Status"
                    },
                    "op": "=",
                    "rhs": {
                        "kind": "Primitive",
                        "text": "Pending",
                        "type": "string"
                    }
                },
                "op": "and",
                "rhs": {
                    "kind": "Search",
                    "text": "international",
                    "value": {
                        "kind": "Primitive",
                        "text": "international",
                        "type": "string"
                    }
                }
            },
            "kind": "Filter"
        }
    ]
}
```

Thanks @mattnibs! This issue can live on to continue tracking the fix to the originally-cited, central issue.
